packages/shared/src/components/HeaderRankProgress.tsx

```diff
@@ -63,7 +63,7 @@ export default function HeaderRankProgress({
   }
 
   const showWelcome = onboardingStep === 1;
-  const showRankAnimation = levelUp && neverShowRankModal;
+  const showRankAnimation = levelUp && !shouldShowRankModal;
```

Comment:
Yes, but this part requires a different condition, it will be the option when we should not show the actual modal, but the animation only, which is why it was used separately on a separate component.